tests/utils/test_io.py

```diff
@@ -33,3 +36,32 @@ def test_valid_filenames(self):
                 else:
                     with self.assertRaises(ValueError):
                         check_filenames(filenames)
+
+class TestEmbeddingsIO(unittest.TestCase):
+
+    def test_valid_embeddings(self):
+        # correct embedding file as created through lightly
+        embeddings_path = tempfile.mktemp('.csv', 'embeddings')
+        embeddings = np.random.rand(32, 2)
+        labels = [0 for i in range(len(embeddings))]
+        filenames = [f'img_{i}.jpg' for i in range(len(embeddings))]
+        save_embeddings(embeddings_path, embeddings, labels, filenames)
+        check_embeddings(embeddings_path)
```

Comment:
Please put this in a separate subtest
`with self.subTest("valid embeddings file"):`